backend/src/services/wallet_service.py

```
@@ -50,6 +42,25 @@ def _initialize_wallet(self):
+            return self.token_decimals_cache[mint_address]
+
+        try:
+            mint_pubkey = Pubkey.from_string(mint_address)
+            response = await self.solana_client.get_account_info(mint_pubkey)
+            if response and response.value:
+                data = response.value.data
+                if len(data) >= 45:
+                    decimals = data[44]
+                    self.token_decimals_cache[mint_address] = decimals
+                    return decimals
+        except Exception as e:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)